From 1dce490fdf829a0decb544a6147c4c5cc4cecf51 Mon Sep 17 00:00:00 2001
From: Quentin Armitage <quentin@armitage.org.uk>
Date: Fri, 14 Jul 2017 16:45:02 +0100
Subject: [PATCH 8/8] Resolve compiler warning introduced by commit 8361b11

Signed-off-by: Quentin Armitage <quentin@armitage.org.uk>
---
 keepalived/check/check_tcp.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/keepalived/check/check_tcp.c b/keepalived/check/check_tcp.c
index 7afed022..69615544 100644
--- a/keepalived/check/check_tcp.c
+++ b/keepalived/check/check_tcp.c
@@ -65,9 +65,6 @@ dump_tcp_check(void *data)
 static bool
 tcp_check_compare(void *a, void *b)
 {
-	tcp_check_t *old = CHECKER_DATA(a);
-	tcp_check_t *new = CHECKER_DATA(b);
-
 	if (!compare_conn_opts(CHECKER_CO(a), CHECKER_CO(b)))
 		return false;
 
-- 
2.20.1

