From ebd51b5f4a26619a4dd0393a7e97e4ca1a122700 Mon Sep 17 00:00:00 2001
From: YAMAMOTO Masaya <pandax381@gmail.com>
Date: Tue, 11 Jul 2017 17:55:50 +0900
Subject: [PATCH 5/8] Set active if new failed_checkers is empty

---
 keepalived/check/ipwrapper.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/keepalived/check/ipwrapper.c b/keepalived/check/ipwrapper.c
index 09d86f08..c9dce2b7 100644
--- a/keepalived/check/ipwrapper.c
+++ b/keepalived/check/ipwrapper.c
@@ -658,6 +658,9 @@ migrate_failed_checkers(real_server_t *old_rs, real_server_t *new_rs)
 			}
 		}
 	}
+
+	if (LIST_ISEMPTY(new_rs->failed_checkers))
+		SET_ALIVE(new_rs);
 end:
 	free_list(&l);
 }
-- 
2.20.1

